src/catchup/CatchupWork.cpp

```diff
@@ -315,9 +321,8 @@ CatchupWork::doWork()
     }
 
     // Step 4.1: Download and verify ledger chain
-    downloadVerifyLedgerChain(catchupRange,
-                              LedgerNumHashPair(catchupRange.first.last(),
-                                                resolvedConfiguration.hash()));
+    downloadVerifyLedgerChain(LedgerNumHashPair(
```

Comment:
it's strange to me that we have a function that verifies a ledger chain range but only takes the end as a parameter. I preferred the old contract.

Maybe it makes more sense to not add `mCatchupRange` at all:
you added this write tests that check that `CatchupWork` is setting things up properly but those tests are not really doing that either.

This can done is a simpler way:
add a method that returns the parameters used for the child work (`CatchupWork` can query `mVerifyLedgers` for the range getting verified, etc).

Reply:
mCatchupRange was also added to avoid recomputing it every time `CatchupWork::doWork` is called (which is a lot).

What do you mean by `those tests are not really doing that either`?

Reply:
> mCatchupRange was also added to avoid recomputing it every time CatchupWork::doWork is called (which is a lot).

I really doubt that this would show up on any profile: constructing this struct requires a handful of comparisons and arithmetic operations.

> What do you mean by those tests are not really doing that either?

what the tests are doing are assuming that child work will indeed be constructed with `mCatchupRange` so it's a relatively weak check.